src/plugins/dashboard/public/saved_dashboards/saved_dashboard.ts

```diff
@@ -77,8 +80,19 @@ export function createSavedDashboardClass(
         type: SavedDashboard.type,
         mapping: SavedDashboard.mapping,
         searchSource: SavedDashboard.searchSource,
-        extractReferences,
-        injectReferences,
+        extractReferences: (opts: {
+          attributes: SavedObjectAttributes;
+          references: SavedObjectReference[];
+        }) => extractReferences(opts, { embeddablePersistableStateService: embeddableStart }),
+        injectReferences: (so: SavedObjectDashboard, references: SavedObjectReference[]) => {
+          const newAttributes = injectReferences(
+            { attributes: so._serialize().attributes, references },
+            {
+              embeddablePersistableStateService: embeddableStart,
+            }
+          );
+          Object.assign(so, newAttributes);
```

Comment:
Not sure about this, but this seem to be working.
I had to decouple `injectReferences` from client side `DashboardSavedObject` class to reuse this on server.